sdk/identity/Azure.Identity/src/SentinalFileLock.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.Identity
+{
+    internal class SentinalFileLock : IDisposable
+    {
+        private FileStream _lockFileStream;
+
+        private SentinalFileLock(FileStream lockFileStream)
+        {
+            _lockFileStream = lockFileStream;
+        }
+        public static async Task<SentinalFileLock> AquireAsync(string lockfilePath, int lockFileRetryCount, TimeSpan lockFileRetryDelay)
+        {
+            Exception exception = null;
+            FileStream fileStream = null;
+
+            // Create lock file dir if it doesn't already exist
+            Directory.CreateDirectory(Path.GetDirectoryName(lockfilePath));
+
+            for (int tryCount = 0; tryCount < lockFileRetryCount; tryCount++)
+            {
+                try
+                {
+                    // We are using the file locking to synchronize the store, do not allow multiple writers for the file.
+                    // Note: this only works on windows if we extend to work on unix systems we need to set FileShare.None
+                    fileStream = new FileStream(lockfilePath, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.Read, 4096, FileOptions.DeleteOnClose);
```

Comment:
Shouldn't it be FileMode.Truncate?

Reply:
I used the same FileMode as was used in the MSAL extensions library which VS is using.